### PR TITLE
department_create_api

### PR DESCRIPTION
**Pull request by varungadde99**
- Rpsec acceptance spec for create
 - Department create service

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,5 +3,6 @@
   api_version(:module => "Api::V1", :header => {
     name: "Accept", :value => "application/vnd.providesk; version=1"}) do
       resources :sessions, only: :create
+      resources :departments
```
Comment (Selectus2) [inline config/routes.rb:6]: there is only one method in departments controller, so avoid creating unnecessary routes

File: config/locales/en.yml [hunk 2]
```diff
@@ -37,6 +37,12 @@ en:
     success: "Logged in successfully"
     failure: "Invalid Credentials"
   hello: "Hello world"
+  department:
+    success:
+      create: "Department created successfully"
+    error:
+      create: "Unable to create department"
+    invalid_params: "Invalid params."
```
Comment (Selectus2) [inline config/locales/en.yml:45]: why it is not inside error ?

File: spec/factories/organizations.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 FactoryBot.define do
   factory :organization do
-    
+    name { "GMAIL"}
```
Comment (Selectus2) [inline spec/factories/organizations.rb:3]: use faker instead of constant strings

File: spec/helpers/login_helper.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+module LoginHelper
```
Comment (Selectus2) [inline spec/helpers/login_helper.rb:1]: Shubham has already written one api headers support helper. how many we are going to create

File: spec/acceptance/v1/departments_spec.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      example 'Unable to create department due to invalid params' do
+        request = {}
+        do_request(request)
+        response_data = JSON.parse(response_body)
+        expect(response_status).to eq(422)
+        expect(response_data["message"]).to eq(I18n.t('department.invalid_params'))
+      end
+
+
+      example 'Unable to create department as department already exisits' do
```
Comment (Selectus2) [inline spec/acceptance/v1/departments_spec.rb:33]: This should be 

> Fail to create a department with the same name

**Review by Selectus2 — CHANGES_REQUESTED** [review]

File: spec/helpers/login_helper.rb (hunk 4 above)
Comment (varungadde99) [reply spec/helpers/login_helper.rb:1]: Ok as disucssed - sticking with Shubham's approach.

File: spec/factories/organizations.rb (hunk 3 above)
Comment (varungadde99) [reply spec/factories/organizations.rb:3]: Sure.

**Review by Selectus2 — APPROVED** [review]
LGTM

**Pull request merged**